pyccel/parser/base.py

```diff
@@ -334,6 +338,28 @@ def blocking(self):
     def show_traceback(self):
         return self._show_traceback
 
+    @property
+    def used_names(self):
+        return self._used_names
+
+    def get_new_name(self, current_name = None):
+        if current_name is not None and current_name not in self._used_names:
+            self._used_names.add(current_name)
+            return current_name
+
+        if current_name is not None:
+            new_name, self._dummy_counter = create_incremented_string(self._used_names, prefix = current_name, counter = self._dummy_counter)
+        else:
+            new_name,_ = create_incremented_string(self._used_names, prefix = current_name)
+        return new_name
+
+    def get_new_variable(self, prefix = None):
```

Comment:
Here too

Reply:
See above! We'll create some automatic documentation for developers at some point, so...